src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -952,14 +951,30 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                         for (i = 0; (uint)i < maxIters; i++)
                         {
                             utf16Data = Unsafe.ReadUnaligned<Vector128<short>>(pInputBuffer);
-                            if (!Sse41.TestZ(utf16Data, nonAsciiUtf16DataMask))
+
+                            if (AdvSimd.IsSupported)
                             {
-                                goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
-                            }
+                                Vector128<short> isUtf16DataNonAscii = AdvSimd.CompareTest(utf16Data, nonAsciiUtf16DataMask);
+                                bool hasNonAsciiDataInVector = AdvSimd.Arm64.MinAcross(isUtf16DataNonAscii).ToScalar() != 0;
+                                if (hasNonAsciiDataInVector)
+                                {
+                                    goto LoopTerminatedDueToNonAsciiDataInVectorLocal;
+                                }
 
-                            // narrow and write
+                                Vector64<sbyte> lower = AdvSimd.ExtractNarrowingSaturateLower(utf16Data);
+                                Vector128<sbyte> result = AdvSimd.ExtractNarrowingSaturateUpper(lower, utf16Data);
+                                AdvSimd.Store((ulong*)pOutputBuffer, result.AsUInt64());
```

Comment:
I'm wondering if this should actually be StoreSelectedScalar with an index of 0? @eiriktsarpalis @kunalspathak

Reply:
I don't see a corresponding code for these lines in x64 case - the latter only does this check with TestZ and goto. Was this added by mistake or I am missing something here?

Reply:
Never mind, apparently, I can't read:)

Reply:
You probably don't need a call to ExtractNarrowingSaturateUpper.

Also I noticed you are using ExtractNarrowingSaturateLower not ExtractNarrowingUnsignedSaturateLower, which one is needed by the algorithm.

If you need to narrow with saturation (up to byte.MaxValue) 8 ushorts and store the resulting 8 bytes to memory then you can use this
```c#
Vector64<byte> lower = AdvSimd.ExtractNarrowingUnsignedSaturateLower(utf16Data);
AdvSimd.Store(pOutputBuffer, lower);
```
Just make sure that the semantics are the same.

Reply:
Thanks! The name is actually `ExtractNarrowingSaturateUnsignedLower`. I got concerned for a second thinking the _unsigned_ method didn't exist 😮.

Reply:
> Thanks! The name is actually `ExtractNarrowingSaturateUnsignedLower`. I got concerned for a second thinking the _unsigned_ method didn't exist 😮.

Ops, got this wrong